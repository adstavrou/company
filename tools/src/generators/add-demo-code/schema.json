{
  "$schema": "http://json-schema.org/schema",
  "title": "Add Demo Generator",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "The folder name for the new demo (e.g. buttons-demo)"
    },
    "title": {
      "type": "string",
      "description": "The display title for the demo (e.g. Buttons Demo)"
    }
  }
}