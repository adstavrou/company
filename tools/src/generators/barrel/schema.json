{
    "$schema": "http://json-schema.org/schema",
    "title": "Barrel Generator",
    "type": "object",
    "properties": {
        "project": {
            "type": "string",
            "description": "The name of the project (library) to generate the barrel for"
        }
    },
    "required": [
        "project"
    ]
}