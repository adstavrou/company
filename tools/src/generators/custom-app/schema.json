{
  "$schema": "http://json-schema.org/schema",
  "title": "Custom App Generator",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "The name of the app"
    }
  }
}