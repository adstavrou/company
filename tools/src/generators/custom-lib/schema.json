{
    "$schema": "http://json-schema.org/schema",
    "title": "Custom Lib Generator",
    "type": "object",
    "properties": {
        "name": {
            "type": "string",
            "description": "Library name"
        },
        "build": {
            "type": "boolean",
            "description": "Add build target?"
        },
        "unitTest": {
            "type": "boolean",
            "description": "Add unit tests?"
        }
    },
    "required": []
}