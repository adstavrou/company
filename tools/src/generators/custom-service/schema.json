{
    "$schema": "http://json-schema.org/schema",
    "title": "Custom Service Generator",
    "type": "object",
    "properties": {
      "name": {
        "type": "string",
        "description": "Name of the service"
      },
      "directory": {
        "type": "string",
        "description": "Target directory inside libs/"
      }
    },
    "required": []
  }