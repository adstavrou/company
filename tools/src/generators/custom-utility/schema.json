{
    "$schema": "http://json-schema.org/schema",
    "title": "Custom Utility Generator",
    "type": "object",
    "properties": {
        "name": {
            "type": "string",
            "description": "The name of the utility function"
        },
        "directory": {
            "type": "string",
            "description": "The library folder (e.g. shared, core-utils)"
        },
        "withTest": {
            "type": "boolean",
            "description": "Generate test file"
        }
    },
    "required": []
}