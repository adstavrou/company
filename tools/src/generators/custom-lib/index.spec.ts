import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';
import { Tree } from '@nx/devkit';
import generator from './index';
import { CustomLibGeneratorSchema } from './schema';

describe('custom-lib generator (snapshot)', () => {
  let tree: Tree;

  beforeEach(() => {
    tree = createTreeWithEmptyWorkspace();
  });

  it('should generate basic lib files and match snapshot', async () => {
    const options: CustomLibGeneratorSchema = {
      name: 'math-utils',
      build: false,
      unitTest: false,
      files: ['constants', 'utils'],
    };

    await generator(tree, options);

    expect(tree.read('libs/math-utils/src/index.ts', 'utf-8')).toMatchSnapshot();
    expect(tree.read('libs/math-utils/src/constants.ts', 'utf-8')).toMatchSnapshot();
    expect(tree.read('libs/math-utils/src/utils.ts', 'utf-8')).toMatchSnapshot();
    expect(tree.read('libs/math-utils/project.json', 'utf-8')).toMatchSnapshot();
  });
});
