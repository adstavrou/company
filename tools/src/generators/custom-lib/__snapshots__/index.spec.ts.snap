// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`custom-lib generator (snapshot) should generate basic lib files and match snapshot 1`] = `
"export const hello = () => console.log('Hello from math-utils!');
"
`;

exports[`custom-lib generator (snapshot) should generate basic lib files and match snapshot 2`] = `
"// Constants for math-utils

export const EXAMPLE_CONSTANT = 'value';
"
`;

exports[`custom-lib generator (snapshot) should generate basic lib files and match snapshot 3`] = `
"// Utility functions

export function exampleUtil() {
  return 'Hello from utils';
}
"
`;

exports[`custom-lib generator (snapshot) should generate basic lib files and match snapshot 4`] = `
"{
  "name": "math-utils",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/math-utils/src",
  "projectType": "library",
  "targets": {},
  "tags": []
}
"
`;
